src/EarlyFree.cpp

```diff
@@ -109,39 +109,28 @@ class FindLastUse : public IRVisitor {
     }
 };
 
-class InjectMarker : public IRMutator {
-public:
-    string func;
-    Stmt last_use;
-
-private:
+Stmt inject_marker(const Stmt &stmt, const string &func, const Stmt &last_use) {
     bool injected = false;
+    return mutate_with(stmt, [&](auto *self, const Block *block) -> Stmt {
```

Comment:
do_injection could take a const Stmt & (an inefficiency, if it is one, also exists in the original). I can't figure out why clang-tidy isn't complaining about it already. Is there something I'm missing?